### PR TITLE
feat: aggregate job error messages from failed tasks and parties

### LINKED ISSUE #529: KusciaJob API 聚合任务失败信息
> 此 ISSUE 为 [隐语开源共建计划（SecretFlow Open Source Contribution Plan，简称 SF OSCP）Phase 5 ](https://studio.secretflow.com/activity/yi9qqz2bvslnbqf/detail)任务 ISSUE，欢迎社区开发者参与共建～
> - 认领前，辛苦确认是否完成[报名](https://studio.secretflow.com/activity/yi9qqz2bvslnbqf/detail)～
> - 详细规则：[「代码季」详细活动说明](https://github.com/secretflow/secretflow/issues/1861)
> - 更多任务，可查看 [「OSCP Phase5 Season of Dev」Project](https://github.com/orgs/secretflow/projects/13)
>
> This ISSUE is one of the tasks of the  [SecretFlow Open Source Contribution Plan (referred to as SF OSCP) Phase 5](https://studio.secretflow.com/activity/yi9qqz2bvslnbqf/detail). Welcome to join us in building it together!
> - Before claiming a task, please make sure you have [signed up](https://studio.secretflow.com/activity/yi9qqz2bvslnbqf/detail).
> - Activity rules: [Detailed activity description of "Season of Dev"](https://github.com/secretflow/secretflow/issues/1861)
> - For more tasks, you can check the ["OSCP Phase5 Season of Dev" Project](https://github.com/orgs/secretflow/projects/13).

### 任务介绍
+ 任务名称：KusciaJob API 聚合任务失败信息
+ 技术方向：Kusica/Job
+ 任务难度：进阶🌟🌟
+ 任务预估完成时间：4周
+ 任务 Reviewer：@IceyZhan

### 详细要求
通过 KusciaJob API 查询作业失败信息时，任务的失败信息比较分散，当 Kuscia Job 状态为 KusciaJobFailed，errMsg 可能为空，真正的准确的 errMsg 需要从 Task 中的 Party 中拿到。优化一下任务失败信息的展示，聚合输出给用户。

+ 实现功能：当调用 KusciaJob Query API 时，聚合 Job 内所有 Task 的 ErrMsg 到 Job 的 ErrMsg 中，ErrMsg 具有易读性，能够体现是 Job 的具体某个 Task 的某个 Party 的错误。
+ 测试说明：需要保证单测和CI通过
+ 提交说明：
    - 设计方案：请先准备你的设计思路方案，包括接口设计和功能实现，并在 ISSUE 下评论提交方案，经过 Reviewer 确认设计思路方案后再根据确认后的方案，进行相关的功能开发
    - 功能开发：功能开发过程中代码需符合代码规范：[https://github.com/uber-go/guide/blob/master/style.md](https://github.com/uber-go/guide/blob/master/style.md)
    - 功能提交：功能开发完成后，关联该 ISSUE 并提交代码至 [https://github.com/secretflow/kuscia](https://github.com/secretflow/kuscia)

### 能力要求
+ 对 Kuscia Job有一定了解
+ 熟悉 golang

### 操作说明
+ Kuscia Job概念：[https://www.secretflow.org.cn/zh-CN/docs/kuscia/v0.13.0b0/reference/concepts/kusciajob_cn](https://www.secretflow.org.cn/zh-CN/docs/kuscia/v0.13.0b0/reference/concepts/kusciajob_cn)
+ Kuscia Job API接口：[https://www.secretflow.org.cn/zh-CN/docs/kuscia/v0.13.0b0/reference/apis/kusciajob_cn](https://www.secretflow.org.cn/zh-CN/docs/kuscia/v0.13.0b0/reference/apis/kusciajob_cn)

### PR DESCRIPTION
**Pull request by RooobinYe**
Implement aggregateErrorMessage function to consolidate error information when KusciaJob fails. The function collects error messages from all failed tasks and parties, providing clearer job-level error reporting.

- Add aggregateErrorMessage method to jobService
- Integrate error aggregation into buildJobStatus
- Add comprehensive unit tests covering all edge cases
- Handle empty messages and mixed party states appropriately

Fixes #529

### CODE REVIEW

**Comment by RooobinYe** [comment]
@IceyZhan PTLA

**Review by IceyZhan — APPROVED** [review]
lgtm

File: pkg/kusciaapi/service/job_service_test.go [hunk 1]
```diff
@@ -213,3 +213,150 @@ func TestBuildScheduleConfigForKusciaAPI(t *testing.T) {
...
+			jobPhase: v1alpha1.KusciaJobFailed,
+			tasks: []*kusciaapi.TaskStatus{
+				{
+					Alias: "task1",
+					State: kusciaapi.JobState_Failed.String(),
+					Parties: []*kusciaapi.PartyStatus{
+						{
+							DomainId: "alice",
+							State:    kusciaapi.JobState_Failed.String(),
+							ErrMsg:   "", // 空错误信息应该被忽略
```
Comment (UniqueMarvin) [inline pkg/kusciaapi/service/job_service_test.go]: 注释 建议用英文替代

File: pkg/kusciaapi/service/job_service_test.go [hunk 2]
```diff
@@ -213,3 +213,150 @@ func TestBuildScheduleConfigForKusciaAPI(t *testing.T) {
 		})
 	}
 }
+
+func TestAggregateErrorMessage(t *testing.T) {
+	// 创建 jobService 实例用于测试
```
Comment (UniqueMarvin) [inline pkg/kusciaapi/service/job_service_test.go]: 注释 建议用英文替代

File: pkg/kusciaapi/service/job_service_test.go [hunk 3]
```diff
@@ -213,3 +213,150 @@ func TestBuildScheduleConfigForKusciaAPI(t *testing.T) {
...
+					Parties: []*kusciaapi.PartyStatus{
+						{
+							DomainId: "alice",
+							State:    kusciaapi.JobState_Failed.String(),
+							ErrMsg:   "connection timeout",
+						},
+						{
+							DomainId: "bob",
+							State:    kusciaapi.JobState_Succeeded.String(),
+							ErrMsg:   "", // 成功的party不应该有错误信息
```
Comment (UniqueMarvin) [inline pkg/kusciaapi/service/job_service_test.go]: 还有这里

**Review by UniqueMarvin — APPROVED** [review]
LGTM
